### PR TITLE
Log: Progress reporter interface

### PR DESCRIPTION
**Pull request by medikoo**
Addresses serverless/serverless#9860

I've decided to use same event emitter as one internally used in `log` engine. It ensures that interface remains environment agnostic.
Not sure if it'll apply that much to _progress_ reporting of which I hardly see a use case beyond Node.js based CLI, still as it's exported together with logging mechanism, it's good to ensure no env specific API's are involved.

### CODE REVIEW

File: lib/log/get-progress-reporter.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const ensureString = require('type/string/ensure');
+const progressEmitter = require('event-emitter')();
+const memoizee = require('memoizee');
+const logLevels = require('log/levels');
+
+module.exports = memoizee(
+  (namespace) => {
```
Comment (pgrzesik) [inline lib/log/get-progress-reporter.js:9]: Should we somehow validate the `namespace` to be a string instance?

File: lib/log/get-progress-reporter.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    return {
+      get: memoizee(
+        (name) => {
```
Comment (pgrzesik) [inline lib/log/get-progress-reporter.js:12]: Same for `name`

**Review by pgrzesik — COMMENTED** [review]
Just a minor comment, looks good otherwise :+1:

File: lib/log/get-progress-reporter.js (hunk 1 above)
Comment (medikoo) [reply lib/log/get-progress-reporter.js:9]: This one is expected only to be called internally (it's not part of public API), it's the reason I decided to not add any validation

File: lib/log/get-progress-reporter.js (hunk 2 above)
Comment (medikoo) [reply lib/log/get-progress-reporter.js:12]: It is validated, see below line :)

Comment (pgrzesik) [reply lib/log/get-progress-reporter.js:12]: right!

**Review by pgrzesik — APPROVED** [review]
Looks great :+1:

**Pull request merged**